devicetypes/timevalve-gaslock-t-08/timevalve-smart.src/timevalve-smart.groovy

```diff
@@ -0,0 +1,281 @@
+                state "battery", label:'${currentValue}% battery', unit:""
+            }
+
+            main (["statusTile"])
+//            details (["statusTile", "remainingSliderTile", "setRemaining", "setTimeout10", "setTimeout20", "batteryTile", "refreshTile", "setTimeout30", "setTimeout40"])
+//            details (["statusTile", "batteryTile", "setRemaining", "refreshTile"])
+            details (["statusTile", "batteryTile", "refreshTile"])
+        }
+}
+
+def parse(description) {
+//	log.debug "parse - " + description
+	def result = null
+	if (description.startsWith("Err 106")) {
+		state.sec = 0
+		result = createEvent(descriptionText: description, isStateChange: true)
+	} else if (description != "updated") {
+		def cmd = zwave.parse(description, [0x20: 1, 0x25: 1, 0x70: 1, 0x71: 1, 0x98: 1])
+		if (cmd) {
+        	log.debug "parsed cmd = " + cmd
+			result = zwaveEvent(cmd)
+			//log.debug("'$description' parsed to $result")
+		} else {
+			log.debug("Couldn't zwave.parse '$description'")
+		}
+	}
+	return result
+}
+
+// 복호화 후 zwaveEvent() 호출
+def zwaveEvent(physicalgraph.zwave.commands.securityv1.SecurityMessageEncapsulation cmd) {
+	//log.debug "SecurityMessageEncapsulation - " + cmd
+	def encapsulatedCommand = cmd.encapsulatedCommand([0x20: 1, 0x25: 1, 0x70: 1, 0x71: 1, 0x98: 1])
+	if (encapsulatedCommand) {
+		state.sec = 1
+        log.debug "encapsulatedCommand = " + encapsulatedCommand
+        zwaveEvent(encapsulatedCommand)
+	}
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.switchbinaryv1.SwitchBinaryReport cmd) {
+	//log.debug "switch status - " + cmd.value
+    createEvent(name:"contact", value: cmd.value ? "open" : "closed")
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.batteryv1.BatteryReport cmd) {
+    def map = [ name: "battery", unit: "%" ]
+    if (cmd.batteryLevel == 0xFF) {  // Special value for low battery alert
+        map.value = 1
+        map.descriptionText = "${device.displayName} has a low battery"
+        map.isStateChange = true
+    } else {
+        map.value = cmd.batteryLevel
+    }
+    
+    log.debug "battery - ${map.value}${map.unit}"
+    // Store time of last battery update so we don't ask every wakeup, see WakeUpNotification handler
+    state.lastbatt = new Date().time
+    createEvent(map)
+}
+
+def zwaveEvent(physicalgraph.zwave.Command cmd) {
+  //log.debug "zwaveEvent - ${device.displayName}: ${cmd}"
+  createEvent(descriptionText: "${device.displayName}: ${cmd}")
+}
+
+def zwaveEvent(physicalgraph.zwave.commands.configurationv1.ConfigurationReport cmd) {
+	def result = []
+	log.info "zwave.configurationV1.configurationGet - " + cmd
+    def array = cmd.configurationValue
+    def value = ( (array[0] * 0x1000000) + (array[1] * 0x10000) + (array[2] * 0x100) + array[3] ).intdiv(60)
+    if (device.currentValue("switch") == "on") {
+    	value = ( (array[0] * 0x1000000) + (array[1] * 0x10000) + (array[2] * 0x100) + array[3] ).intdiv(60)
+    } else {
+    	value = 0
+    }
```

Comment:
Actually, this can all just be consolidated to 

    def value = 0
    if (device.currentValue("contact") == "open") {
    	value = ( (array[0] * 0x1000000) + (array[1] * 0x10000) + (array[2] * 0x100) + array[3] ).intdiv(60)

    	def hour = value.intdiv(60);
        def min = (value % 60).toString().padLeft(2, '0');
        def text = "${hour}:${min}M"

...